### PR TITLE
Remove downwards dependency from `HUDOverlay` to `HealthDisplay`

### PR DESCRIPTION
**Pull request by peppy**
Preparation for skinning customisation.

### CODE REVIEW

File: osu.Game/Screens/Play/HUD/HealthDisplay.cs [hunk 1]
```diff
@@ -29,12 +32,18 @@ protected virtual void Flash(JudgementResult result)
...
+        private void load(HUDOverlay hud)
         {
             Current.BindTo(HealthProcessor.Health);
 
             HealthProcessor.NewJudgement += onNewJudgement;
+
+            if (hud != null)
+            {
+                showHealthbar = hud.ShowHealthbar.GetBoundCopy();
+                showHealthbar.BindValueChanged(healthBar => this.FadeTo(healthBar.NewValue ? 1 : 0, HUDOverlay.FADE_DURATION, HUDOverlay.FADE_EASING), true);
```
Comment (bdach) [inline osu.Game/Screens/Play/HUD/HealthDisplay.cs:45]: That's a `BindValueChanged(..., true)` in BDL - I'm not sure whether oversight or intentional, though, given this becomes a bit uglier if the callback was to be conditionally invoked in `LoadComplete()`? Doesn't crash on the transforms at least.

File: osu.Game/Screens/Play/HUD/HealthDisplay.cs [hunk 2]
```diff
@@ -29,12 +32,18 @@ protected virtual void Flash(JudgementResult result)
         {
         }
 
-        [BackgroundDependencyLoader]
-        private void load()
+        [BackgroundDependencyLoader(true)]
+        private void load(HUDOverlay hud)
```
Comment (bdach) [inline osu.Game/Screens/Play/HUD/HealthDisplay.cs:36]: Could annotate the param (and field, possibly) as `[CanBeNull]`.

File: osu.Game/Screens/Play/HUD/HealthDisplay.cs (hunk 1 above)
Comment (peppy) [reply osu.Game/Screens/Play/HUD/HealthDisplay.cs:45]: While it's probably safe as both components are added together, but I'll move it anyways.

File: osu.Game/Screens/Play/HUD/HealthDisplay.cs (hunk 2 above)
Comment (peppy) [reply osu.Game/Screens/Play/HUD/HealthDisplay.cs:36]: If we're gonna start doing this we probably want an inspection rule or something. It's only done in a handful of places we use the `CanBeNull` param.

**Review by smoogipoo — APPROVED** [review]

**Pull request merged**